task_processing/plugins/mesos/execution_framework.py

```diff
@@ -138,8 +161,28 @@ def kill_tasks_stuck_in_staging(self):
                         )
                         get_metric(TASK_STUCK_COUNT).count(1)
 
+            self._reconcile_tasks()
             time.sleep(10)
 
+    def _reconcile_tasks(self):
+        if time.time() < self._reconcile_tasks_at:
```

Comment:
not a blocker, but long term I'd like the decision of 'when' to reconcile to be the responsibility of the caller, so that we can have an implicit reconciliation on both a regular period basis (every 15 mins) as well as the explicit reconciliation required when we're asking mesos about specific tasks.

Reply:
Sounds good. The current setup is for implicit invocation. We can change this if there is a need for explicit invocation.